src/OpenTelemetry/Metrics/ExportMode.cs

```diff
@@ -0,0 +1,47 @@
+// <copyright file="ExportMode.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+
+namespace OpenTelemetry.Metrics
+{
+    [Flags]
+    public enum ExportMode : byte
```

Comment:
I think typically enums with `[Flags]` have plural names.

Reply:
Should this be a flags? Is `ExportMode.Push | ExportMode.Pull` valid?

Reply:
I think yes, refer to https://github.com/open-telemetry/opentelemetry-specification/pull/1888#discussion_r704053279.

Reply:
> I think typically enums with `[Flags]` have plural names.

I struggled with the name, especially for `AggregationTemporality` (I don't want to call it `AggregationTemporalities` because 1) it's hard to search 2) it's not something used by the spec)

Reply:
My personal practice is to always end non-flags enums with "Type" eg `AggregationTemporalityType`. Break it sometimes, but it helps me to start there.

Reply:
`AggregationTemporality` is now a flag, I made the change https://github.com/open-telemetry/opentelemetry-dotnet/blob/3cc0f7e4af03a49942b4e2b75d383b73b91e2ad6/src/OpenTelemetry/Metrics/AggregationTemporality.cs#L22.

Reply:
In that case same feedback as `ExportMode` in that IMO it should be plural. I think either `AggregationTemporalityTypes` or `AggregationTemporalityModes` could work. Probably lots of options 😄 `AggregationTemporalityStyles`?

Reply:
+1 on `AggregationTemporalityModes` to align with `ExportModes` (btw, do we want it to be `ExportingModes` - I personally would prefer not having `-ing`).
-1 on `AggregationTemporalityTypes` and `AggregationTemporalityStyles` due to the misalignment with `ExportModes` (`ExportTypes` and `ExportStyles` sound awkward to me).

Reply:
I've updated the PR to use `ExportModes`. Will send a separate PR to address `AggregationTemporalityModes`.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet/pull/2357/files#r709700649